resource/systemd/states.go

```diff
@@ -0,0 +1,186 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.package systemd
+
+package systemd
+
+import (
+	"strings"
+
+	"github.com/coreos/go-systemd/dbus"
+	godbus "github.com/godbus/dbus"
+)
+
+func propDependency(name string, units []string) *dbus.Property {
+	return &dbus.Property{
+		Name:  name,
+		Value: godbus.MakeVariant(units),
+	}
+}
+
+type LoadState string
+type LoadStates []LoadState
+
+const (
+	LSLoaded   LoadState = "loaded"
+	LSError    LoadState = "error"
+	LSMasked   LoadState = "masked"
+	LSNotFound LoadState = "not-found"
+)
+
+func PropLoadState(ls LoadState) *dbus.Property {
+	return &dbus.Property{
+		Name:  "LoadState",
+		Value: godbus.MakeVariant(string(ls)),
+	}
+}
+
+func (states LoadStates) Contains(state LoadState) bool {
+	for _, s := range states {
+		if s == state {
+			return true
+		}
+	}
+	return false
+}
+
+type ActiveState string
+type ActiveStates []ActiveState
+
+const (
+	ASActive       ActiveState = "active"
+	ASReloading    ActiveState = "reloading"
+	ASInactive     ActiveState = "inactive"
+	ASFailed       ActiveState = "failed"
+	ASActivating   ActiveState = "activating"
+	ASDeactivating ActiveState = "deactivating"
+)
+
+func (state ActiveState) Equal(state2 ActiveState) bool {
+	if state == state2 {
+		return true
+	}
+	state = ActiveState(strings.Replace(string(state), "\"", "", -1))
```

Comment:
are you trying to remove quotes here? Seems like you ought to remove them higher up the stack so you don't ever get an invalid value in a state. If you *do* have to do it here though: `strings.Trim(string(state), "\"") == strings.Trim(string(state), "\"")` works just fine.

Oh, and it might read better if `state2` was called `other`.